s3/pkg/datastore/gcp/gcp.go

```diff
@@ -46,9 +46,66 @@ type GcsAdapter struct {
 	session *s3client.Client
 }
 
+func (ad *GcsAdapter) BucketDelete(ctx context.Context, in *pb.Bucket) error {
+	log.Info("Bucket delete is called in gcp")
+	Region := aws.String(ad.backend.Region)
+	Endpoint := aws.String(ad.backend.Endpoint)
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      Region,
+		Endpoint:    Endpoint,
+		Credentials: Credentials,
+	}
+
+	svc := awss3.New(session.New(configuration))
+
+	input := &awss3.DeleteBucketInput{
+		Bucket: aws.String(in.Name),
+	}
+
+	result, err := svc.DeleteBucketWithContext(ctx, input)
```

Comment:
Put a comment as in why DeleteBucketWithContext is used nit simple DeletBucket

Reply:
included